File: solutions/dart/hamming/lib/hamming.dart
import 'package:collection/collection.dart';

const errInvalidLength = 'strands must be of equal length';

class Hamming {
  int distance(String left, String right) {
    if (left.length != right.length) {
      throw new ArgumentError(errInvalidLength);
    }

    return new IterableZip([left.runes, right.runes])
        .where((runes) => runes.first != runes.last)
        .length;
  }
}

File: solutions/csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
            throw new ArgumentException("strands must have the same length.");

        return firstStrand.Zip(secondStrand).Count(paired => paired.First != paired.Second);
    }
}
